### PR TITLE
Log warning from included TOC

### PR DESCRIPTION
**Pull request by superyyrrzz**
#3026

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs [hunk 1]
```diff
@@ -86,6 +100,18 @@ private static string ParseFile(string link)
             return queryIndex == -1 ? link : link.Remove(queryIndex);
         }
 
+        private static void AddOrUpdate(Dictionary<string, ImmutableList<LinkSourceInfo>> dict, string path, LinkSourceInfo source)
+            => dict[path] = dict.TryGetValue(path, out var sources) ? sources.Add(source) : ImmutableList.Create(source);
+
+        private static LinkSourceInfo GetLinkSourceInfo(string path, string anchor, string source, string includedFrom)
+        {
+            return new LinkSourceInfo()
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: >() [](start = 37, length = 2)

no need

File: src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs [hunk 2]
```diff
@@ -53,31 +55,43 @@ private void BuildCore(TocItemViewModel item, FileModel model, IHostService host
...
                 foreach (var i in item.Items)
                 {
-                    BuildCore(i, model, hostService);
+                    BuildCore(i, model, hostService, includedFrom);
                 }
             }
+
+            void UpdateDependencies(HashSet<string> linkTos, Dictionary<string, ImmutableList<LinkSourceInfo>> linkSources, string link)
+            {
+                var path = UriUtility.GetPath(link);
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: >UriUtility.GetPath(link) [](start = 27, length = 24)

can this handle uid?

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs]: yes, the implementation simply split by `#`

---
In reply to: [208089474](https://github.com/dotnet/docfx/pull/3106#discussion_r208089474) [](ancestors = 208089474)

File: test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs [hunk 3]
```diff
@@ -774,6 +774,36 @@ public void LoadTocYamlWithEmptyNodeShouldSucceed()
...
+        {
+            // Arrange
+            var masterContent = @"
+- name: TOC2
+  href: ../included/toc.yml";
+            var includedContent = @"
+- name: Article2
+  href: not-existing2.md";
+            var files = new FileCollection(_inputFolder);
+            var masterFile = _fileCreator.CreateFile(masterContent, FileType.YamlToc, "master");
```
Comment (herohua) [inline test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs]: add a case for `uid`

Comment (superyyrrzz) [reply test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocDocumentProcessorTest.cs]: added and fixed the result...

**Pull request merged**